api/src/websocket/controllers/logs.ts

```diff
@@ -0,0 +1,67 @@
+import { useEnv } from '@directus/env';
+import type { Server as httpServer } from 'http';
+import type WebSocket from 'ws';
+import emitter from '../../emitter.js';
+import { useLogger } from '../../logger/index.js';
+import { refreshAccountability } from '../authenticate.js';
+import { handleWebSocketError } from '../errors.js';
+import { AuthMode, WebSocketMessage } from '../messages.js';
+import type { AuthenticationState, WebSocketClient } from '../types.js';
+import SocketController from './base.js';
+
+const logger = useLogger();
+
+export class LogsController extends SocketController {
+	constructor(httpServer: httpServer) {
+		super(httpServer, 'WEBSOCKETS_LOGS');
+
+		const env = useEnv();
+
+		this.server.on('connection', (ws: WebSocket, auth: AuthenticationState) => {
```

Comment:
We should check the accountability here for `admin=true` and close the connection if not. Or try to do that a level higher in the base controller before this `connection` event gets called.

Reply:
Added in https://github.com/directus/directus/pull/23016/commits/ed0c0fc2fce21e58ed7a072ddfaeec3e8d52283c 👍